library/core/src/main/java/com/google/android/exoplayer2/video/MediaCodecVideoRenderer.java

```diff
@@ -1055,6 +1055,10 @@ protected MediaFormat getMediaFormat(
     mediaFormat.setInteger(MediaFormat.KEY_HEIGHT, format.height);
     MediaFormatUtil.setCsdBuffers(mediaFormat, format.initializationData);
     // Set format parameters that may be unset.
+    if(MediaCodecUtil.getCodecProfileAndLevel(format.codecs) != null) {
```

Comment:
Dolby Vision implementation of platforms may expose multiple codecs each supporting a profile. Ideally I would like to see the MediaCodecSelector doing the selection work and returning the correct codec. But looks that MediaCodecSelector just selects based on mime-type only.

Please suggest a better method if any.

Reply:
We've been planning to propagate the profile information to the MediaCodecSelector for a while but haven't got round to it yet. This is also needed for picking the right VP9 video codec (hardware/software) for HDR vs non-HDR profiles.

Would be okay to support only the 'one DV decoder' case in this change, and I'll make a later commit that handles this in a way that works for arbitrary codecs where we know profile capabilities?

Reply:
Ok. I will make the change.